handler/handler.go

```diff
@@ -18,6 +24,43 @@ type NatsQueue struct {
 	Topic     string
 }
 
+func (q *NatsQueue) connect() error {
+	nc, err := stan.Connect(
+		q.ClusterID,
+		q.ClientID,
+		stan.NatsURL(q.NATSURL),
+		stan.SetConnectionLostHandler(func(conn stan.Conn, err error) {
+			log.Printf("Disconnected from %s\n", q.NATSURL)
+
+			q.reconnect(0)
+		}),
+	)
+
+	if err != nil {
+		return err
+	}
+
+	q.nc = nc
+
+	return nil
+}
+
+func (q *NatsQueue) reconnect(iteration int) {
+	log.Printf("Trying to reconnect (%d) to %s\n", iteration, q.NATSURL)
+
+	if iteration < maxReconnect {
+		time.Sleep(time.Second * time.Duration(iteration * delayBetweenReconnect))
+
+		if err := q.connect(); err != nil {
+			log.Printf("Reconnection (%d) to %s failed", iteration, q.NATSURL)
+
+			q.reconnect(iteration + 1)
+		} else {
+			log.Printf("Reconnection (%d) to %s succed", iteration, q.NATSURL)
+		}
+	}
```

Comment:
You may want to log when reaching the max reconnect since you are then giving up in connecting to the server.

Reply:
Done